pages/settings/index.html

```diff
@@ -120,6 +120,52 @@ <h3 data-string="settingsPasswordAutoFillHeadline"></h3>
 
 	</div>
 
+	<div class="settings-container" id="proxy-settings-container">
+		<h3 data-string="settingsProxyHeading"></h3>
+
+		<div class="setting-section" id="proxy-type-container">
+			<select id="selected-proxy-type" style="min-width: 225px">
+				<option data-string="settingsNoProxy"></option>
+				<option data-string="settingsManualProxy"></option>
+				<option data-string="settingsAutomaticProxy"></option>
+			</select>
+		</div>
+
+		<div class="setting-section" id="proxy-grid">
+			<div class="setting-option">
+				<label for="http-proxy-host-input">HTTP</label>
+				<input id="http-proxy-host-input" type="text" name="httpHost" placeholder="proxy.example.com" />
+				<input type="number" id="http-proxy-port-input" name="httpPort" min="0" placeholder="8080" />
+			</div>
+			<div class="setting-option">
+				<label for="https-proxy-host-input">HTTPS</label>
+				<input id="https-proxy-host-input" type="text" name="httpsHost" placeholder="proxy.example.com" />
+				<input type="number" id="https-proxy-port-input" name="httpsPort" min="0" placeholder="8080" />
+			</div>
+			<div class="setting-option">
+				<label for="socks-proxy-host-input">SOCKS</label>
+				<input id="socks-proxy-host-input" type="text" name="socksHost" placeholder="proxy.example.com" />
+				<input type="number" id="socks-proxy-port-input" name="socksPort" min="0" placeholder="8080" />
+			</div>
+			<div class="setting-option">
+				<input id="socks4-input" type="radio" value="socks4" name="socksVersion" />
+				<label for="socks4-input">SOCKS v4</label>
+				<input id="socks5-input" type="radio" value="socks5" name="socksVersion" />
+				<label for="socks4-input">SOCKS v5</label>
+			</div>
```

Comment:
If I'm understanding correctly, this could all just be combined into a single text input, so you could type in "http://example.com:8080" directly, which would simplify this code a lot (like I mentioned here: https://github.com/minbrowser/min/pull/820#issuecomment-529302002)

Related to this, it's not clear to me what happens if you fill out more than one of the field sets, so there should probably be some UI or explanation for that. But if there's only a single text input, it would only be possible to set one proxy, so that issue would go away.